### ISSUE TITLE
[HttpClient] base_url path is replaced instead of appended to

### ISSUE THREAD

**Issue by silverbackdan** [symfony/symfony#39724]
**Symfony version(s) affected**: 5.2

**Description**  
If I set the `base_uri` to `http://example.com/api` in a client, then when I make a request e.g. `/resource` (or `resource` without the slash) the URL becomes `http://example.com/resource` instead of `http://example.com/api/resource`

**How to reproduce**  
Set the base_uri in a scoped client (or any client) and then make a request as detailed above.

**Possible Solution**  
The base is used if not path is specified in the new request as seen here https://github.com/symfony/http-client/blob/5.x/HttpClientTrait.php#L409

This line will strip the `/api` instead of including it
https://github.com/symfony/http-client/blob/5.x/HttpClientTrait.php#L417

Currently I would have to include the trailing slash in the `base_uri` as well to make this all work. e.g. `http://example.com/api/`

**Additional context**  
It may be nice to document (if it isn't already and I just missed it) that using a base_url and then defining a path beginning with a slash will replace the path in the base_uri. E.g. using `/resource` will result in `http://example.com/resource` and `resource` will become `http://example.com/api/resource`

**Comment by nicolas-grekas** [comment]
If you want the path to be kept, you must add a trailing slash.
`http://example.com/api/`

**Comment by silverbackdan** [comment]
Am I missing this info in the docs or would a docs PR be appreciated? I've had a search through and couldn't see info on the trailing slash or prefix slash on the paths but I could have missed it.

**Comment by nicolas-grekas** [comment]
A doc PR would definitely be appreciated.

**Comment by jderusse** [comment]
this is documented here https://symfony.com/doc/current/reference/configuration/framework.html#base-uri

**Comment by realjjaveweb** [comment]
@nicolas-grekas I still don't think this is how people expect it to work, now I would understand if adding/removing trailing slash to the base URI _**would**_ actually make the difference, **_but it doesn't,_** because usually the calling is written as `/some/thing` which, as currently documented, also results in removing the path form `base_uri`. If it's problematic to resolve this due to backwards compatibility issues, I would suggest using some special placeholder in the base_uri path that would later be removed in the process but at the same time it would be a way how to ensure 100% the path stays in the base_uri. e.g.<br>`[KEEP_PATH]https://my-base-uri.host/my/base/path`

**TL;DR Main point is:** editing `base_uri` does **not** give you 100% control over the base **path** inclusion/exclusion...<br>And that's just wrong.

**Comment by nicolas-grekas** [comment]
As mentioned in the doc, we follow [RFC 3986](https://www.ietf.org/rfc/rfc3986.txt).
I'm not sure what's not possible, but we're not going to diverge from the standard for sure.
